lib/marked.js

```diff
@@ -505,7 +505,7 @@ inline._punctuation = '!"#$%&\'()*+,\\-./:;<=>?@\\[^_{|}~';
 inline.em = edit(inline.em).replace(/punctuation/g, inline._punctuation).getRegex();
 
 inline._inside = /(?:\[[^\]]*\]|[^\[\]]|\](?=[^\[]*\]))*/;
-inline._href = /(?:[^()]|\([^()]*\))*/;
+inline._href = /(?:[^()]|\([^()]*\)|\S*\(?\([^()]*\)\))*/;
```

Comment:
A few things about this regex:
1. Why have the `\S*` in the third part of the regex? I don't think we should need whitespace before nested brackets.
2. Why make the first opening bracket optional with `\(?`? It should probably just be `\(` since this otherwise allows for mismatched brackets
3. The third part of the regex doesn't allow for things like `(abc(123))` or `((abc)123(xyz))` where there are multiple sets of nested brackets.

I think this might work
```suggestion
inline._href = /(?:[^()]|\([^()]*\)|\((?:[^()]*\([^()]*\))+[^()]*\))*/;
```
I think this might work since the third group would now match
```
\(             // an outer set of brackets containing
(?:
    [^()]*     // some text
    \(         // followed by matching brackets
    [^()]*     // containing some more text
    \)
)+             // repeated any number of times
[^()]*         // and then finally followed by some more text
\)
```